Run RPC JIT tests with variable type hints only in Python >=3.6

xush6528 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34284 Run RPC JIT tests with variable type hints only in Python >=3.6**

Python 3.5 only supports function type hints.
Variable type hints are introduced in Python 3.6.
So these tests with JIT type hints will fail with "Syntax Error" in Python 3.5 environment.

Differential Revision: [D7348891](https://our.internmc.facebook.com/intern/diff/D7348891/)

xush6528 committed:
Run RPC JIT tests with type hints in only in Python >=3.6

`

Differential Revision: [D7348891](https://our.internmc.facebook.com/intern/diff/D7348891/)

[ghstack-poisoned]

mrshenli reviewed (commented):
There might be a conflict with this one. Probably need a rebase

https://github.com/pytorch/pytorch/commit/1546d2afeb98fcd7bc5b58261d6e31ad7794e833

xush6528 committed:
Update on "Run RPC JIT tests with variable type hints only in Python >=3.6"


Python 3.5 only supports function type hints.
Variable type hints are introduced in Python 3.6.
So these tests with JIT type hints will fail with "Syntax Error" in Python 3.5 environment.

Differential Revision: [D7348891](https://our.internmc.facebook.com/intern/diff/D7348891/)

[ghstack-poisoned]
